mri/parallel_mri/extract_sensitivity_maps.py

```diff
@@ -178,19 +173,20 @@ def get_Smaps(k_space, img_shape, samples=None, mode='Gridding',
         fourier_op = NFFT(samples=samples, shape=img_shape)
         Smaps = np.asarray([fourier_op.adj_op(k_space[l]) for l in range(L)])
     else:
-        xi = np.asarray(range(img_shape[0]))
-        yi = np.asarray(range(img_shape[1]))
-        gridx, gridy = np.meshgrid(xi, yi)
-        for l in range(L):
-            gridded_kspace = gridding_2d(
-                samples,
-                k_space[l],
-                img_shape,
-                method=method,
-                point_min=min_samples,
-                point_max=max_samples)
-            Smaps[l] = pfft.fftshift(pfft.ifft2(pfft.ifftshift(
-                gridded_kspace))).T
+        Smaps = []
+        grid_space = [np.linspace(min_samples[i],
+                                  max_samples[i],
+                                  num=img_shape[i],
+                                  endpoint=False)
+                      for i in np.arange(np.size(img_shape))]
```

Comment:
This might make it fail silently if the mininum samples do not have a shape corresponding to the image dimension (imagine a 2D image and the default values you have). It might be better to check the size.

Reply:
Yes, I think I'll not have any default arguments, we should not assume the dimension especially when we didn't assume for img_shape.